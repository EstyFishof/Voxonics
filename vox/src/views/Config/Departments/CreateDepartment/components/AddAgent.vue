<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>ADD AGENTS</h2>
      <span class="hr"></span>
    </div>
    <div class="top">
      <div class="left">Add agent
      </div>
      <div class="agents users">
        <div v-if="newDepartmentAgents.length" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentAgents.length > 1" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentAgents.length > 2" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentAgents.length > 3" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
      </div>
    </div>
    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <div class="label" v-click-outside="hideListUsers">
              <input placeholder="Enter name or Agent Number"
                     class="input-field"
                     @click="showList()"
                     @input="showAgents = true; searchDepartments()"
                     v-model.trim="name"/>
              <span @click="showList()" class="triangle" :class="{'rotate-arrow': showAgents}">
                <img src="@/assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <DropdownList v-show="showAgents" :classNameList="'list'" :dataList="mainList.getMainList"  @scrollFunc="(event) => funScroll(event)"
                  @clickFunc="(item) => chooseAgent(item)" :itemText="(item) => `<span>${item.name}</span><span>ID ${item.internalNumber}</span>`"
                  :classNameItemComputed="(item) => newDepartmentAgents.filter(element => element.uid === item.uid).length ? collapse ? 'selected collapsed row' : 'selected row' : 'row'"
                  :showItem="(item) => newDepartmentAgents.filter(element => element.uid === item.uid).length ? !collapse : true">
                <span class="collapse" @click="collapse = !collapse">
                  <img class="button" :src="collapse ? extendImg : collapseImg" alt="collapse">
                  <span class="text">Selected users available to you:</span>
                  <span class="amount">{{ mainList.topList.length }}</span>
                </span>
              </DropdownList>
            </div>
          </div>
        </div>
      </div>
      <div class="agents users">
        <div class="agent user" v-for="(agent, index) of newDepartmentAgents" :key="index">
          <div class="block">
            <div class="round"><img :src="agent.avatar || userAva" alt="avatar"></div>
            <span class="name">{{ agent.name }}</span>
          </div>
          <span class="name">{{ agent.internalNumber }}</span>
          <img @click="chooseAgent(agent)" class="union" src="@/assets/img/icons/union.svg" width="19" alt="">
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import userAva from '@/assets/img/icons/user_icon.svg'
import collapseImg from '@/assets/img/icons/collapse.svg'
import extendImg from '@/assets/img/icons/extend.svg'
import DropdownList from '@/components/DropdownList'
import ClickOutside from "vue-click-outside"
import {dropdownListFetch, dropdownChooseItem} from '@/lib/dropdownListsDepsAndQueues.js'
import _ from 'lodash'

export default {
  name: "AddAgent",
  components: {
    DropdownList
  },
  data() {
    return {
      collapseImg,
      extendImg,
      collapse: false,
      name: "",
      showAgents: false,
      userAva: userAva,
      perPage: 9,
      mainList: {
        topList: [],
        bottomList: [],
        getMainList: []
      },
    }
  },
  watch: {
    getGlobalUsersList() {
      dropdownListFetch(this, this.getGlobalUsersList, 'newDepartmentAgents')
      this.mainList.getMainList = _.concat(this.mainList.topList, this.mainList.bottomList)
    }
  },
  computed: {
    ...mapGetters(['newDepartmentAgents', 'getGlobalUsersList', 'getPaginatorUsers']),
  },
  methods: {
    searchDepartments() {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch(), 800)
    },
    funScroll({target}) {
      if(((target.clientHeight + target.scrollTop === target.scrollHeight) && (this.getPaginatorUsers.paginator.count > this.perPage))) {
        this.perPage += 9
        this.fetch()
      }
    },
    fetch() {
      this.$store.dispatch('globalUsersListRequest', {peerPage: this.perPage, sort: '', page: 1, search: this.name})
    },
    showList() {
      if(!this.showAgents) {
        this.fetch()
      }
      this.showAgents = !this.showAgents
    },
    hideListUsers() {
      this.showAgents = false
    },
    chooseAgent(agent) {
      dropdownChooseItem(this, agent, 'newDepartmentAgents', 'setAgentsDepartment')
    },
    setAgentsDepartment(array) {
      this.$store.dispatch('transferAgentsDepartment', array)
    }
  },
  mounted() {
    this.popupItem = this.$el.children[2].children[0]
    if (this.newDepartmentAgents.length) {
      this.mainList.topList = this.newDepartmentAgents
    }
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/agentsSettings.scss"></style>

<style scoped lang="scss">
.personal-info-wrap {
  background: none;
  padding-left: 0;
  border-top: none;
  height: 363px;

  .wrap-head {
    h2 {
      font-size: 48px;
      margin-right: 24px;
    }
  }
  
  .top {
    margin: 50px 24px 0 24px;

    .left {
      margin-right: 44px;
    }
  }

  .wrap-body {
    margin: 0 0 36px 0;

    .personal-info {

      &__first-part {
        .item-container {
          margin: 0 24px 16px 24px;

          .label {
            .list {
              height: 175px;
            }
          } 
        }
      }
    }
  }

  .agents {
    .agent {
      color: var(--font-panel);
    }
  }
}
@media (max-width: 1780px) {
  .personal-info-wrap {
    .top {
      margin: 50px 0 0 0;

      .left {
        margin-right: 20px;
        margin-left: 24px;
      }
    }
    .wrap-body {
      .personal-info__first-part {
        .item-container {
          margin: 0 0 16px 24px;
        }
      }
    }
  }
}
</style>
