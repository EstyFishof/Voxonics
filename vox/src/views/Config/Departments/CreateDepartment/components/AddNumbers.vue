<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>ASSIGN V-NUMBERS</h2>
      <span class="hr"></span>
    </div>
    <div class="top">
      <div class="left">Assign number
      </div>
      <div class="numbers">
        <div v-if="newDepartmentNumbers.length" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentNumbers.length > 1" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentNumbers.length > 2" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentNumbers.length > 3" class="number">
          <div class="block">
            <div class="round">№ <span class="vnumber">VNumber</span></div>
            <span class="name"></span>
          </div>
          <span class="name"></span>
          <span class="union"></span>
        </div>
      </div>
    </div>
    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <div class="label" v-click-outside="hideListNumbers">
              <input placeholder="Enter or select number"
                     class="input-field"
                     @click="showList()"
                     @input="showNumbers = true; searchNumbers()"
                     v-model.trim="name"/>
              <span @click="showList()" class="triangle" :class="{'rotate-arrow': showNumbers}">
                <img src="@/assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <DropdownList v-show="showNumbers" :classNameList="'list'" :dataList="mainList.getMainList"  @scrollFunc="(event) => funScroll(event)"
                  @clickFunc="(item) => chooseNumber(item)" :itemText="(item) => `<span>${item.description} ${item.number}</span>`"
                  :classNameItemComputed="(item) => newDepartmentNumbers.filter(element => element.uid === item.uid).length ? 'selected row' : 'row'" />
            </div>
          </div>
        </div>
      </div>
      <div class="numbers">
        <div class="number" v-for="(number, index) of newDepartmentNumbers" :key="index">
          <div class="block">
            <div class="round">{{ index + 1 }}.</div>
            <span class="name">{{ number.number }}</span>
            <span class="description">{{ number.description }}</span>
          </div>
          <span class="name">{{ number.internalNumber }}</span>
          <img @click="chooseNumber(number)" class="union" src="@/assets/img/icons/union.svg" width="19" alt="">
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import ClickOutside from "vue-click-outside"
import DropdownList from '@/components/DropdownList'
import {dropdownListFetch, dropdownChooseItem} from '@/lib/dropdownListsDepsAndQueues.js'
import _ from 'lodash'

export default {
  name: "AddVNumbers",
  components: {
    DropdownList
  },
  data() {
    return {
      name: "",
      showNumbers: false,
      perPage: 12,
      mainList: {
        topList: [],
        bottomList: [],
        getMainList: []
      },
    }
  },
  watch: {
    getGlobalExternalNumbers() {
      dropdownListFetch(this, this.getGlobalExternalNumbers, 'newDepartmentNumbers')
      this.mainList.getMainList = _.concat(this.mainList.topList, this.mainList.bottomList)
    }
  },
  computed: {
    ...mapGetters(['newDepartmentNumbers', 'getGlobalExternalNumbers', 'getPaginatorNumbers']),
  },
  methods: {
    searchNumbers() {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch(), 800)
    },
    funScroll({target}) {
      if((target.clientHeight + target.scrollTop === target.scrollHeight)) {
        if(this.getPaginatorNumbers.paginator.count > this.perPage) {
          this.perPage += 9
          this.fetch()
        }
      }
    },
    fetch() {
      this.$store.dispatch('setGlobalExternalNumbers', {peerPage: this.perPage, number: this.name})
    },
    showList() {
      this.fetch()
      this.showNumbers = !this.showNumbers
    },
    hideListNumbers() {
      this.showNumbers = false
    },
    chooseNumber(num) {
      dropdownChooseItem(this, num, 'newDepartmentNumbers', 'setNumbersDepartment')
    },
    setNumbersDepartment(array) {
      this.$store.dispatch('transferNumberDepartment', array)
    }
  },
  mounted() {
    this.popupItem = this.$el.children[2].children[0]
    if (this.newDepartmentNumbers.length) {
      this.mainList.topList = this.newDepartmentNumbers
    }
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/agentsSettings.scss"></style>
<style scoped lang="scss">
.personal-info-wrap {
  background: none;
  padding-left: 0;
  border-top: none;
  height: 363px;

  .wrap-head {
    h2 {
      font-size: 48px;
      margin-right: 24px;
    }
  }
  .top {
    margin: 50px 24px 0 24px;
    .number {
      .vnumber {
        margin-left: 20px;
      }
    }
  }

  .wrap-body {
    margin: 0 0 36px 0;

    .personal-info {

      &__first-part {
        .item-container {
          margin: 0 24px 16px 24px;
          .label {
            .list {
              height: 175px;
            }
          } 
        }
      }
    }
  }

  .numbers {
    .number {
      .block {
        position: relative;

        .round {
          border-radius: 0;
          margin-right: 0;
          border: none;
        }

        .description {
          position: absolute;
          left: 25px;
          top: 21px;
          font-weight: 300;
          font-size: 10px;
          line-height: 9px;
          color: #8F8F8F;
        }
      }
    }
  }
}
@media (max-width: 1780px) {
  .personal-info-wrap {
    .top {
      margin: 50px 0 0 0;
      .left {
        margin-right: 20px;
        margin-left: 24px;
      }
    }
    .wrap-body {
      .personal-info__first-part {
        .item-container {
          margin: 0 0 16px 24px;
        }
      }
    }
  }
}
</style>
