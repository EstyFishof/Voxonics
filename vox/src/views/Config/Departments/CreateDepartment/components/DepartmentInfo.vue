<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>DEPARTMENT INFO</h2>
      <span class="hr"></span>
    </div>

    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <label>
              Name
              <input placeholder="Enter name of Department"
                     class="input-field"
                     v-model.trim="name"
                     maxlength="12"
                     @change="setDepartmentName"
                     required
                     :class="{error: requiredName}"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"

export default {
  name: "DepartmentInfo",
  props: ['requiredName'],
  data() {
    return {
      name: "",
    }
  },
  computed: {
    ...mapGetters(['newDepartmentName'])
  },
  methods: {
    setDepartmentName() {
      this.$store.dispatch('transferNameDepartment', this.name)
    }
  },
  mounted() {
    if (this.newDepartmentName) {
      this.name = this.newDepartmentName
    }
  }
}
</script>

<style scoped lang="scss">
.personal-info-wrap {
  .wrap-body {
    display: flex;
    flex-direction: row;
    margin: 36px 0 36px 24px;

    .personal-info {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;

      &__first-part {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;

        .item-container {
          margin: 16px 0;

          label {
            font-size: 18px;
            font-weight: 700;
            text-align: left;
            color: var(--light-blue);

            .error {
              border: 1px solid red !important;
              outline: red;
            }
          }
        }
      }
    }
  }

  .with-err {
    position: relative;
    margin-bottom: 30px;
  }

}
</style>
