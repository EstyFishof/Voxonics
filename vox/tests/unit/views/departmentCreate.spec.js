import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Department from '../../../src/views/Config/Departments/CreateDepartment/CreateDepartment.vue';
import AddAgent from '../../../src/views/Config/Departments/CreateDepartment/components/AddAgent.vue';
import AddManagers from '../../../src/views/Config/Departments/CreateDepartment/components/AddManagers.vue';
import DepartmentInfo from '../../../src/views/Config/Departments/CreateDepartment/components/DepartmentInfo.vue';
import AddNumbers from '../../../src/views/Config/Departments/CreateDepartment/components/AddNumbers.vue';
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
const router = new VueRouter();
localVue.use(Vuex);

describe('CreateDepartment', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

const store = new Vuex.Store({
    state: {
        dep: {relatedWith: {externalNumberUid: [], userUid: []}, belongTo: []},
        userInfo: {
            option: {
                edit: false, 
                internalNumber: 136,
            },
            permission: {
                role: 'maganer'
            },
            internalNumber: 124
        }
    },
    // effects: {
    //     setAgents(state, value) {
    //         state.queue = value
    //     }
    // },
    getters: {
        newDepartment(state) {
            return state.dep;
        },
        userInfo(state) {
            return state.userInfo;
        }
    }
});
    const wrapper = shallowMount(Department, { store, localVue, router, stubs: {Department: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('handlerSubmit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'handlerSubmit');
        await wrapper.vm.handlerSubmit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', () => {
        Department.mounted.call(wrapper.vm);
    });
});

describe('AddAgent', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            agentsDep: [{}, {}],
            agents: [],
            userInfo: {
                option: {
                    edit: false, 
                    internalNumber: 136,
                },
                permission: {
                    role: 'maganer'
                },
                internalNumber: 124
            }
        },
        actions: {
            transferAgentsDepartment({commit}, value) {},
        },
        mutations: {
            setDepAgents(state, value) {
                state.agentsDep = value;
            }
        },
        getters: {
            newDepartmentAgents(state) {
                return state.agentsDep;
            },
            agents(state) {
                return state.agents;
            },
            userInfo(state) {
                return state.userInfo;
            }
        }
    });
    const wrapper = shallowMount(AddAgent, { store, localVue, router, stubs: {AddAgent: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.vm.chooseAgent({});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.setData({relatedAgents: [{uid: 111}]});
        await wrapper.vm.chooseAgent({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setAgentsDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setAgentsDepartment');
        await wrapper.vm.setAgentsDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', () => {
    //     AddAgent.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: 'aaa'})
    //     AddAgent.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: '123'})
    //     AddAgent.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        const wrapper = shallowMount(AddAgent, { store, localVue, router });
        await AddAgent.mounted.call(wrapper.vm);
    });
});

describe('AddManagers', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            managersDep: [{option: {edit: true}}],
            managers: [],
            userInfo: {
                option: {
                    edit: false, 
                    internalNumber: 136,
                },
                permission: {
                    role: 'maganer'
                },
                internalNumber: 124
            },
            newDepartmentManagers: [
                {
                    avatar: '',
                    name: '',
                    internalNumber: '',
                    option: {
                        edit: false
                    }
                }
            ]
        },
        actions: {
            transferManagersDepartment({commit}, value) {},
        },
        mutations: {
            setQueueManagers(state, value) {
                state.managersDep = value;
            }
        },
        getters: {
            newDepartmentManagers(state) {
                return state.managersDep;
            },
            managers(state) {
                return state.managers;
            },
            userInfo(state) {
                return state.userInfo;
            }
        }
    });
    const wrapper = shallowMount(AddManagers, { store, localVue, router, stubs: {AddManagers: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await store.commit('setQueueManagers', []);
        await wrapper.vm.chooseAgent({});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.setData({relatedManagers: [{uid: 111}]});
        await wrapper.vm.chooseAgent({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setAgentsDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setAgentsDepartment');
        await wrapper.vm.setAgentsDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: 'aaa'})
    //     AddManagers.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: '123'})
    //     AddManagers.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        const wrapper = shallowMount(AddManagers, { store, localVue, router });
        await AddManagers.mounted.call(wrapper.vm);
    });
});

describe('DepartmentInfo', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            // dep: {relatedWith: {externalNumberUid: [], userUid: []}, belongTo: []}
            name: 'dep',
            userInfo: {
                option: {
                    edit: false, 
                    internalNumber: 136,
                },
                permission: {
                    role: 'maganer'
                },
                internalNumber: 124
            }
        },
        actions: {
            transferNameDepartment({commit}, value) {},
        },
        mutations: {
            nameMutations(state, value) {
                state.queueName = value;
            },
        },
        getters: {
            newDepartmentName(state) {
                return state.name;
            },
            userInfo(state) {
                return state.userInfo;
            }
        }
    });
    const wrapper = shallowMount(DepartmentInfo, { store, localVue, router, stubs: {DepartmentInfo: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('setDepartmentName toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setDepartmentName');
        await wrapper.vm.setDepartmentName();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(DepartmentInfo, { store, localVue, router });
        await DepartmentInfo.mounted.call(wrapper.vm);
    });
});

describe('AddNumbers', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            numbers: [],
            depNumbers: [],
            userInfo: {
                option: {
                    edit: false, 
                    internalNumber: 136,
                },
                permission: {
                    role: 'maganer'
                },
                internalNumber: 124
            }
        },
        actions: {
            transferNumberDepartment({commit}, value) {},
        },
        mutations: {
            depNumbersMutations(state, value) {
                state.depNumbers = value;
            },
        },
        getters: {
            numbers(state) {
                return state.numbers;
            },
            newDepartmentNumbers(state) {
                return state.depNumbers;
            },
            userInfo(state) {
                return state.userInfo;
            }
        }
    });
    const wrapper = shallowMount(AddNumbers, { store, localVue, router, stubs: {AddNumbers: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseNumber');
        await wrapper.setData({relatedNumbers: [{uid: 222}]});
        await wrapper.vm.chooseNumber({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseNumber');
        await wrapper.setData({relatedNumbers: [{uid: 111}]});
        await wrapper.vm.chooseNumber({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setNumbersDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setNumbersDepartment');
        await wrapper.vm.setNumbersDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', async () => {
    //     await AddNumbers.computed.numbersFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', async () => {
    //     wrapper.setData({name: 'aaa'})
    //     await AddNumbers.computed.numbersFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', async () => {
    //     wrapper.setData({name: '123'})
    //     await AddNumbers.computed.numbersFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        await AddNumbers.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        store.commit('depNumbersMutations', [{}, {}]);
        await AddNumbers.mounted.call(wrapper.vm);
    });
});


