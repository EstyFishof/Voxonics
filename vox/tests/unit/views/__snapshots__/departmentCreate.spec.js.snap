// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAgent Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      ADD AGENTS
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add agent

    </div>
     
    <div
      class="agents users"
    >
      <div
        class="agent user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            Name
          </div>
           
          <span
            class="name"
          />
        </div>
         
        <span
          class="name"
        >
          Agent Number
        </span>
         
        <span
          class="union"
        />
      </div>
       
      <div
        class="agent user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            Name
          </div>
           
          <span
            class="name"
          />
        </div>
         
        <span
          class="name"
        >
          Agent Number
        </span>
         
        <span
          class="union"
        />
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="agents users"
    >
      <div
        class="agent user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            <img
              alt="avatar"
              src=""
            />
          </div>
           
          <span
            class="name"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
      <div
        class="agent user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            <img
              alt="avatar"
              src=""
            />
          </div>
           
          <span
            class="name"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AddManagers Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Managed by
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add manager

    </div>
     
    <div
      class="managers users"
    >
      <div
        class="manager user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            Name
          </div>
           
          <span
            class="name"
          />
        </div>
         
        <span
          class="name"
        >
          Agent Number
        </span>
         
        <span
          class="union"
        />
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="managers users"
    >
      <div
        class="manager user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            <img
              alt="avatar"
              src=""
            />
          </div>
           
          <span
            class="name"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <div
          class="action"
        >
          <label
            class="label-container"
          >
            <input
              checked="checked"
              class="view"
              disabled="disabled"
              type="checkbox"
            />
             
            <span
              class="check"
            />
             
            <span
              class="text"
            >
              View
            </span>
          </label>
           
          <label
            class="label-container"
          >
            <input
              class="edit"
              type="checkbox"
            />
             
            <span
              class="check"
            />
             
            <span
              class="text"
            >
              Edit
            </span>
          </label>
        </div>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AddNumbers Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      ASSIGN V-NUMBERS
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Assign number
      
    </div>
     
    <div
      class="numbers"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter or select number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="numbers"
    />
  </div>
</div>
`;

exports[`CreateDepartment Check snapshot 1`] = `
<div
  class="wrap-main"
>
  <div
    class="wrap-form"
  >
    <form>
      <departmentinfo-stub />
       
      <addagent-stub />
       
      <addmanager-stub />
       
      <addvnumbers-stub />
       
      <div
        class="btn-container"
      >
        <button
          class="btn-submit"
        >
          Apply Changes
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`DepartmentInfo Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      DEPARTMENT INFO
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <label>
            
              Name
 
            <input
              class="input-field"
              maxlength="12"
              placeholder="Enter name of Department"
              required="required"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
